From: Pascal Nowack <Pascal.Nowack@gmx.de>
Date: Tue, 12 Apr 2022 08:56:45 +0200
Subject: vnc-pipewire-stream: Only advertise dma-buf support when available

Use the previously implemented API in the EGL thread to determine,
whether dma-bufs can be used or not.

https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/issues/89
---
 src/grd-vnc-pipewire-stream.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/grd-vnc-pipewire-stream.c b/src/grd-vnc-pipewire-stream.c
index cc1a29b..6ef7bee 100644
--- a/src/grd-vnc-pipewire-stream.c
+++ b/src/grd-vnc-pipewire-stream.c
@@ -194,6 +194,7 @@ on_stream_param_changed (void                 *user_data,
   GrdVncPipeWireStream *stream = GRD_VNC_PIPEWIRE_STREAM (user_data);
   GrdSession *session = GRD_SESSION (stream->session);
   GrdContext *context = grd_session_get_context (session);
+  GrdEglThread *egl_thread = grd_context_get_egl_thread (context);
   uint8_t params_buffer[1024];
   struct spa_pod_builder pod_builder;
   int width;
@@ -217,7 +218,7 @@ on_stream_param_changed (void                 *user_data,
   grd_session_vnc_queue_resize_framebuffer (stream->session, width, height);
 
   allowed_buffer_types = 1 << SPA_DATA_MemFd;
-  if (grd_context_get_egl_thread (context))
+  if (egl_thread && grd_egl_thread_supports_dma_bufs (egl_thread))
     allowed_buffer_types |= 1 << SPA_DATA_DmaBuf;
 
   params[0] = spa_pod_builder_add_object (
@@ -746,7 +747,7 @@ connect_to_stream (GrdVncPipeWireStream  *stream,
   add_common_format_params (&pod_builder, spa_format);
 
   egl_thread = grd_context_get_egl_thread (context);
-  if (egl_thread)
+  if (egl_thread && grd_egl_thread_supports_dma_bufs (egl_thread))
     {
       uint32_t drm_format;
       int n_modifiers;
